src/klein/test/test_resource.py

```diff
@@ -390,6 +390,23 @@ def deferred(request):
         self.assertFired(d)
         self.assertEqual(request.getWrittenData(), b"ok")
 
+    def test_asyncRendering(self):
+        app = self.app
+        resource = self.kr
+
+        request = requestMock(b"/resource/leaf")
+
+        @app.route("/resource/leaf")
+        async def leaf(request):
+            return LeafResource()
+
+        expected = b"I am a leaf in the wind."
+
+        def assertResult(_):
+            self.assertEqual(request.getWrittenData(), expected)
+
+        return _render(resource, request).addCallback(assertResult)
```

Comment:
Hrm. Could you use `successResultOf` here rather than returning a `Deferred`?

Reply:
done